src/dds/rs-dds-auto-calibration.h

```diff
@@ -0,0 +1,49 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <src/auto-calibrated-device.h>
+#include <src/ds/d500/d500-auto-calibration.h>
```

Comment:
This puts the DS dependency front and center, and makes the dependency inescapable.
Can we have this only in the .cpp?
I.e., have dds_device_proxy derive from `auto_calibrated_interface` and the implementation will have the dependency?

Reply:
yep